ocaml/xapi/repository_helpers.ml

```diff
@@ -1076,3 +1076,24 @@ let prune_updateinfo_for_livepatches livepatches updateinfo =
     List.filter (fun x -> LivePatchSet.mem x livepatches) updateinfo.livepatches
   in
   {updateinfo with livepatches= lps}
+
+let do_with_device_models ~__context
+    ?(only_vm_with_recommended_guidances = false) ~host f =
```

Comment:
You can also put the filter in the `match` of `restart_device_models_for_recommended_guidances`.
In this case, the specific logic will be in the specific function.